src/main/java/com/google/sps/ChatWebSocket.java

```diff
@@ -59,9 +64,46 @@ public void onOpen(Session session) throws IOException {
     }
 
     @OnMessage
-    public void textMessage(Session session, String msg) {
+    public void textMessage(Session session, String msg) throws IOException {
+        JSONObject messageJson = new JSONObject(msg);
+
+        switch (messageJson.getString("type")) {
+            case "MSG_SEND":
+                chatMessage(session, messageJson);
+                break;
+            case "MAP_SEND":
+
+                break;
+            default:
+                System.out.println("Invalid message");
```

Comment:
Instead of printing this here,  consider throwing an IllegalArgumentException. When something goes wrong, it's safer to fail early that to continue executing code.